lib/travis/build/script/go.rb

```diff
@@ -19,6 +19,9 @@ class Go < Script
         def export
           super
           sh.export 'TRAVIS_GO_VERSION', go_version, echo: false
+          sh.if '-z $GOMAXPROCS' do
+            sh.export 'GOMAXPROCS', '2', echo: true
```

Comment:
It could be made more cross-platform friendly a la:

```bash
nproc 2>/dev/null || sysctl -a | awk '/machdep\.cpu\.core_count/ { print $NF }' 2>/dev/null || echo 2
```